### PR TITLE
First PR

### PR DESCRIPTION
**Pull request by Raksha120799**

### CODE REVIEW

File: buyGPS/src/main/java/com/TruckBooking/buyGPS/Model/BuyGPSRequest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    private long rate;
+	
+	@NotBlank(message="duration has to be specified")
+    private String duration;
+	
+	@NotBlank(message="address cannot be empty")
+    private String address;
+	
+	@CreatedDate
+    private String purchaseDate;
```
Comment (badireddisatyasaivarun) [inline buyGPS/src/main/java/com/TruckBooking/buyGPS/Model/BuyGPSRequest.java:33]: I don't think purchageDate, installedStatus, imei and timestamp should be present. Refer docs.
Create Request model for post and put seperately.

File: buyGPS/src/main/java/com/TruckBooking/buyGPS/Service/BuyGPSServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+		String temp ="";
+		BuyGPS buygps = new BuyGPS();
+		CreateBuyGPSResponse response = new CreateBuyGPSResponse();
+		
+		temp = "gps:" + UUID.randomUUID();
+		buygps.setGpsId(temp);
+		response.setGpsId(temp);
+		
+		temp = buygpsrequest.getTransporterId();
+		buygps.setTransporterId(temp);
```
Comment (badireddisatyasaivarun) [inline buyGPS/src/main/java/com/TruckBooking/buyGPS/Service/BuyGPSServiceImpl.java:43]: check whether transporterId is null or not, before assigning.
and find out what are optional and compulsory fields.

File: buyGPS/src/main/java/com/TruckBooking/buyGPS/Service/BuyGPSServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+			throw ex;
+		}
+		else if(D.isPresent())
+		{
+			buygpsdao.deleteById(gpsId);
+			deleteGps.setMessage(CommonConstants.deleteSuccess);
+			log.info("delete BuyGPS service has been executed");
+			return deleteGps;
+		}
+		else
```
Comment (badireddisatyasaivarun) [inline buyGPS/src/main/java/com/TruckBooking/buyGPS/Service/BuyGPSServiceImpl.java:228]: I dont think this else should be present.

File: buyGPS/src/main/java/com/TruckBooking/buyGPS/Model/BuyGPSRequest.java (hunk 1 above)
Comment (Raksha120799) [reply buyGPS/src/main/java/com/TruckBooking/buyGPS/Model/BuyGPSRequest.java:33]: You mean purchaseDate, imei, installedStatus and timestamp are not required in Model class?
Sure I will create request model for post and put seperately.

File: buyGPS/src/main/java/com/TruckBooking/buyGPS/Service/BuyGPSServiceImpl.java (hunk 2 above)
Comment (Raksha120799) [reply buyGPS/src/main/java/com/TruckBooking/buyGPS/Service/BuyGPSServiceImpl.java:43]: Sure I will check if it is null or not. In the document for buygps there is no optional or compulsory fields mentioned.

File: buyGPS/src/main/java/com/TruckBooking/buyGPS/Service/BuyGPSServiceImpl.java (hunk 3 above)
Comment (Raksha120799) [reply buyGPS/src/main/java/com/TruckBooking/buyGPS/Service/BuyGPSServiceImpl.java:228]: Ok then I will erase that else. Thanks for reviewing and suggesting changes.

File: buyGPS/src/main/java/com/TruckBooking/buyGPS/Service/BuyGPSServiceImpl.java (hunk 2 above)
Comment (badireddisatyasaivarun) [reply buyGPS/src/main/java/com/TruckBooking/buyGPS/Service/BuyGPSServiceImpl.java:43]: Ask Pranav, which are compulsory fields and optionally fields are.

**Review by liveasy97 — APPROVED** [review]

File: buyGPS/src/main/java/com/TruckBooking/buyGPS/Service/BuyGPSServiceImpl.java (hunk 3 above)
Comment (Raksha120799) [reply buyGPS/src/main/java/com/TruckBooking/buyGPS/Service/BuyGPSServiceImpl.java:228]: Do you want me to just remove the else and the paranthesis or the whole block?

**Pull request merged**